toolchain/diagnostics/diagnostic.h

```diff
@@ -119,8 +119,10 @@ struct DiagnosticBase {
   explicit constexpr DiagnosticBase(DiagnosticKind kind, DiagnosticLevel level,
                                     llvm::StringLiteral format)
       : Kind(kind), Level(level), Format(format) {
-    static_assert((... && !std::is_same_v<Args, llvm::StringRef>),
-                  "Use std::string or llvm::StringLiteral for diagnostics to "
+    static_assert((... && !(std::is_same_v<Args, llvm::StringRef> ||
+                            std::is_same_v<Args, llvm::StringLiteral>)),
+                  "For diagnostics, use a format provider (see "
+                  "toolchain/diagnostics/format_providers.h) or std::string to "
                   "avoid lifetime issues.");
```

Comment:
String literals don't have lifetime issues, so we probably need a separate error message for that.

Reply:
Changed the error message to just point at the diagnostic parameter type advice.